src/server/saved_objects/client/saved_objects_client.js

```diff
@@ -25,6 +24,62 @@ export class SavedObjectsClient {
     this._unwrappedCallCluster = callCluster;
   }
 
+  /**
+   * ## SavedObjectsClient errors
```

Comment:
Alright, you asked for some docs @kjbekkelund :) I don't think it's bad to have some of these thought written down but let me know if you think I went overboard

Reply:
❤️ I like this! It gives tons of context and makes it way easier to understand why we do what we do.